pkg/controller/embercsi/common.go

```diff
@@ -447,7 +447,7 @@ func generateVolumes(ecsi *embercsiv1alpha1.EmberCSI, csiDriverMode string) []co
 			Name: "socket-dir",
 			VolumeSource: corev1.VolumeSource{
 				HostPath: &corev1.HostPathVolumeSource{
-					Path: fmt.Sprintf("%s/%s", "/var/lib/kubelet/plugins", GetPluginDomainName(ecsi.Name)),
+					Path: fmt.Sprintf("%s/%s", "/var/lib/kubelet", GetPluginDomainName(ecsi.Name)),
```

Comment:
I don't think this will work, because we'll be mounting the kubelet directory multiple times, and that will mess the detach process (the same volume will be seen as mounted multiple times inside the container).

I think it would be best to use: `Path: fmt.Sprintf("%s/%s", "/var/lib/ember-csi", GetPluginDomainName(ecsi.Name)),
 `

And we probably want to mount that directory under `/var/lib/ember-csi` so we are storing everything from inside the container there (locks, bind-mounts, socket, etc.)

Here are the changes that were made for the plugin: https://github.com/embercsi/ember-csi-operator/pull/96/files